packages/core/src/logs/db.ts

```diff
@@ -0,0 +1,49 @@
+import { Database } from "bun:sqlite";
+import { join } from "node:path";
+import { type BunSQLiteDatabase, drizzle } from "drizzle-orm/bun-sqlite";
+import * as schema from "./schema.js";
+
+/**
+ * Database connection cache
+ * Key: storage directory path
+ * Value: Drizzle database instance
+ */
+const dbCache = new Map<string, BunSQLiteDatabase<typeof schema>>();
+
+/**
+ * Get or create a Drizzle database connection for the given storage directory.
+ *
+ * Uses singleton pattern - one connection per storage directory.
+ *
+ * @param storageDir - Path to storage directory (e.g., ~/.mcp-gateway/capture)
+ * @returns Drizzle database instance
+ */
+export function getDb(storageDir: string): BunSQLiteDatabase<typeof schema> {
```

Comment:
what is a scenario where one might expect multiple storage directories?

Reply:
I don't think we have one right now. Looking at it a bit more closely, i want to refactor the code because there are so many singletons in the chain of storage and the core package.